tests/dragonfly/json_test.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+from redis.commands.json.path import Path
+from .utility import *
+
+jane = {
+    'name': "Jane",
+    'Age': 33,
+    'Location': "Chawton"
+}
+
+json_num = {
+    "a":{"a":1, "b":2, "c":3}
+}
+
+def get_type(connection, key):
+    return connection.execute_command("type", key)
+
+def get_set_json(connection, key, value, path="$"):
+
+    try:
+        connection.json().set(key, path, value)
+
+        result = connection.json().get(key, path)
+        return result
+    except redis.exceptions.ResponseError as re:
+        return str(re)
+    except Exception as e:
+        return None
+
+def get_str(connect, key):
+    try:
+        return connect.get(key)
+    except redis.exceptions.ResponseError as re:
+        return str(re)
+    except Exception as e:
+        return None
+
+def test_basic_json_get_set(client):
+    key_name="test-json-key"
+    result = get_set_json(connection=client, key=key_name, value=jane)
+    if result is None:
+        assert False, "failed to set JSON value"
+    else:
+        the_type = get_type(connection=client, key=key_name)
+        assert the_type == "ReJSON-RL"
+        assert len(result) == 1
+        assert result[0]['name'] == 'Jane'
+        assert result[0]['Age'] == 33
+
+def test_access_json_value_as_string(client):
+    key_name="test-json-key"
+    result = get_set_json(connection=client, key=key_name, value=jane)
+    if result is None:
+        assert False, "failed to set JSON value"
+    else:
+        # make sure that we have valid JSON here
+        the_type = get_type(connection=client, key=key_name)
+        assert the_type == "ReJSON-RL"
+        # you cannot access this key as string
+        result = get_str(connect=client, key=key_name)
+        assert result is not None
+        assert result == "WRONGTYPE Operation against a key holding the wrong kind of value"
+
+def test_reset_key_to_string(client):
+    key_name="test-json-key"
+    result = get_set_json(connection=client, key=key_name, value=jane)
+    if result is None:
+        assert False, "failed to set JSON value"
+    else:
+        # make sure that we have valid JSON here
+        the_type = get_type(connection=client, key=key_name)
+        assert the_type == "ReJSON-RL"
+        # set the key to be string - this is legal
+        assert client.set(key_name, "some random value"), "we should be able to set type to string"
+        result = get_str(connect=client, key=key_name)
+        assert result == "some random value"
+        # For JSON set the update the root path, we are allowing
+        # to change the type to JSON and override it
+        result = get_set_json(connection=client, key=key_name, value=jane)
+        the_type = get_type(connection=client, key=key_name)
+        assert the_type == "ReJSON-RL"
+
+def test_update_value(client):
+    key_name="test-json-key"
+    result = get_set_json(connection=client, key=key_name, value=json_num)
+    if result is None:
```

Comment:
assert result, "failed to set JSON value"

Reply:
I've changed all these cases, see the comment above
I will add one more test case, to ensure that we're not able to update with JSON.set command when the value is not JSON